docs/installing.rst

```diff
@@ -87,11 +87,16 @@ Then you can build the image using:
 Option 5: Build Solang from source
 ----------------------------------
 
-In order to build Solang from source, you will need rust 1.68.0 or higher,
-and a build of LLVM based on the Solana LLVM tree. There are a few LLVM patches required that are not upstream yet.
+In order to build Solang from source, you will need:
+
+* Rust version 1.68.0 or higher
+* A C++ compiler with support for C++17
+* A build of LLVM based on the Solana LLVM tree. There are a few LLVM patches required that are not upstream yet.
```

Comment:
Our LLVM binary already comes with clang, which can build C++ 17.

Reply:
> Our LLVM binary already comes with clang, which can build C++ 17.

Only the wasm and sbf targets are enabled, not the native target (e.g. intel or arm). So, this compiler won't work. We may need to adjust the build for that. I am guessing the native clang/gcc is being used in CI and @xermicus' local machine

Reply:
That's a very good point. I need to investigate how the `build.rs` of wasm-opt selects the compiler, as we want to make sure that this works consistently

Reply:
@seanyoung `wasm-opt` uses the `cc` crate under the hood, which expects [`cc` on unix](https://github.com/rust-lang/cc-rs#compile-time-requirements) for unix like systems. So on Linux and MacOS it'll always work, unless someone overwrites their `cc` with the solana labs LLVM project. And because that toolchain only supports wasm and SBF, this would represent a severe misconfiguration of their system inducing a lot of problems elsewhere.

For windows it seems fine too as [`cl.exe`](https://learn.microsoft.com/en-us/cpp/build/reference/compiler-options?view=msvc-170) seems to be the equivalent of unix `cc`.